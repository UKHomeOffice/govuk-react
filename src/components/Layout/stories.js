import React from "react";
import { storiesOf } from "@storybook/react";
import Layout from ".";
import GridRow from "../GridRow";
import GridCol from "../GridCol";

storiesOf("Layout", module).add("Column Full", () => (
  <Layout>
    <GridRow>
      <GridCol columFull>
        <p>content</p>
      </GridCol>
    </GridRow>
  </Layout>
));

storiesOf("Layout", module).add("Column Halves", () => (
  <Layout>
    <GridRow>
      <GridCol columnOneHalf>
        <p>content</p>
      </GridCol>
      <GridCol columnOneHalf>
        <p>content</p>
      </GridCol>
    </GridRow>
  </Layout>
));

storiesOf("Layout", module).add("Column Thirds", () => (
  <Layout>
    <GridRow>
      <GridCol columnOneThird>
        <p>content</p>
      </GridCol>
      <GridCol columnOneThird>
        <p>content</p>
      </GridCol>
      <GridCol columnOneThird>
        <p>content</p>
      </GridCol>
    </GridRow>
  </Layout>
));

storiesOf("Layout", module).add("Column Two Thirds / One Third", () => (
  <Layout>
    <GridRow>
      <GridCol columnTwoThirds>
        <p>content</p>
      </GridCol>
      <GridCol columnOneThird>
        <p>content</p>
      </GridCol>
    </GridRow>
  </Layout>
));

storiesOf("Layout", module).add("Column One Third / Two Thirds", () => (
  <Layout>
    <GridRow>
      <GridCol columnOneThird>
        <p>content</p>
      </GridCol>
      <GridCol columnTwoThirds>
        <p>content</p>
      </GridCol>
    </GridRow>
  </Layout>
));

storiesOf("Layout", module).add("Quarters", () => (
  <Layout>
    <GridRow>
      <GridCol columnOneQuarter>
        <p>content</p>
      </GridCol>
      <GridCol columnOneQuarter>
        <p>content</p>
      </GridCol>
      <GridCol columnOneQuarter>
        <p>content</p>
      </GridCol>
      <GridCol columnOneQuarter>
        <p>content</p>
      </GridCol>
    </GridRow>
  </Layout>
));

storiesOf("Layout", module).add("Example layout", () => (
  <Layout>
    <GridRow>
      <GridCol showContent columFull>
        <h2>full column</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data.
        </p>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol showContent columnTwoThirds>
        <h2>Two thirds</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data.
        </p>
      </GridCol>
      <GridCol showContent columnOneThird>
        <h2>One third</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol showContent columnOneThird>
        <h2>One third</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data.
        </p>
      </GridCol>
      <GridCol showContent columnOneThird>
        <h2>One third</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
      <GridCol showContent columnOneThird>
        <h2>One third</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol showContent columnOneHalf>
        <h2>One Half</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data.
        </p>
      </GridCol>
      <GridCol showContent columnOneHalf>
        <h2>One Half</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol showContent columnOneHalf>
        <h2>One Half</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data.
        </p>
      </GridCol>
      <GridCol showContent columnOneQuarter>
        <h2>One quarter</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
      <GridCol showContent columnOneQuarter>
        <h2>One quarter</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
    </GridRow>
    <GridRow>
      <GridCol showContent columnOneQuarter>
        <h2>One quarter</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data.
        </p>
      </GridCol>
      <GridCol showContent columnOneQuarter>
        <h2>One quarter</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
      <GridCol showContent columnOneQuarter>
        <h2>One quarter</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data
        </p>
      </GridCol>
      <GridCol showContent columnOneQuarter>
        <h2>One quarter</h2>
        <p>
          This guide shows how to make your service look consistent with the
          rest of GOV.UK. It includes example code and guidance for layout,
          typography, colour, images, icons, forms, buttons and data.
        </p>
      </GridCol>
    </GridRow>
  </Layout>
));
