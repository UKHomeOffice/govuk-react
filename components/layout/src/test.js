import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import Layout from './';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('Layout', () => {
  let props;
  const example = 'example';
  const wrapper = <Layout>{example}</Layout>;

  beforeEach(() => {
    props = {
      children: example,
    };
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Layout>Example</Layout>, div);
  });

  it('passes `props.children` to the rendered `wrapper` as `children`', () => {
    expect(wrapper.props.children).toBe(props.children);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
