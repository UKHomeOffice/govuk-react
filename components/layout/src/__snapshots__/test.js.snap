// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  padding: 10px;
  box-sizing: border-box;
}

<Layout>
  <src__StyledLayout>
    <div
      className="c0"
    >
      example
    </div>
  </src__StyledLayout>
</Layout>
`;
